### PR TITLE
Move changelog to towncrier

### LINKED ISSUE #230: Switch to towncrier or similar
I’m sick of fuzzing with CHANGELOG.rst.  I *think* we need at least one new release tho to keep up the quality of your changelog.  (notably: links to issues have been added recently and not released I believe @hawkowl?)

Also some fix/workaround for hawkowl/towncrier#77 is necessary.

### PR DESCRIPTION
**Pull request by hynek**
Fixes #230

No changelog merge conflicts anymore!

Does this make sense @hawkowl ?

### CODE REVIEW

File: CONTRIBUTING.rst [hunk 1]
```diff
@@ -86,17 +86,50 @@ Documentation
...
-  Use `semantic newlines`_, and add a link to your pull request:
+
+
+Changelog
+^^^^^^^^^
+
+If your change is noteworthy, there needs to be a changelog entry, so our users can learn about it!
+
+To avoid merge conflicts, we use the towncrier_ package to manage our changelog.
+``towncrier`` uses independent files for each pull request -- so called *news fragments* -- instead one monolithic changelog file.
```
Comment (blueyed) [inline CONTRIBUTING.rst]: s/instead one/instead of one/

File: CONTRIBUTING.rst [hunk 2]
```diff
@@ -86,17 +86,50 @@ Documentation
...
+On release, those news fragments are compiled into our ``CHANGELOG.rst``.
```
Comment (blueyed) [inline CONTRIBUTING.rst]: No comma after release?!

File: CONTRIBUTING.rst [hunk 3]
```diff
@@ -86,17 +86,50 @@ Documentation
...
+
+You don't need to install ``towncrier`` yourself, you just have to abide to a few simple rules:
+
+- For each pull request, add a new file into ``changelog.d`` with a filename adhering to the ``pr#.change|deprecation|breaking.rst`` schema:
```
Comment (blueyed) [inline CONTRIBUTING.rst]: `pr#.change|deprecation|breaking.rst` => `pr#.(change|deprecation|breaking).rst` ?

File: CONTRIBUTING.rst [hunk 4]
```diff
@@ -86,17 +86,50 @@ Documentation
...
+  For example ``changelog.d/42.change.rst`` for a non-breaking change, that is proposed in pull request number 42.
+- As with other docs, please use `semantic newlines`_ within news fragments.
+- Wrap symbols like modules, functions, or classes into double backticks so they are rendered in a monospaced font.
+- If you mention functions or other callables, add parantheses at the end of their names: ``attr.func()`` or ``attr.Class.method()``.
+  This makes the changelog a lot more readable.
+- Prefer simple past or constructions with "now".
+  For example:
+
+  + Added ``attr.validators.func()``.
```
Comment (blueyed) [inline CONTRIBUTING.rst]: s/Added/Add ?!

File: CONTRIBUTING.rst [hunk 5]
```diff
@@ -86,17 +86,50 @@ Documentation
...
 
   .. code-block:: rst
 
-     - Added ``attr.validators.func()``.
-       The feature really *is* awesome.
-       `#1 <https://github.com/python-attrs/attrs/pull/1>`_
-     - ``attr.func()`` now doesn't crash the Large Hadron Collider anymore.
-       The bug really *was* nasty.
-       `#2 <https://github.com/python-attrs/attrs/pull/2>`_
+     Added ``attr.validators.func()``.
```
Comment (blueyed) [inline CONTRIBUTING.rst]: s/Added/Add ?

File: CONTRIBUTING.rst (hunk 4 above)
Comment (hynek) [inline CONTRIBUTING.rst]: I’ve moved away from present tense in changelogs.  It reads weird and is kind of just the remnant of using present tense in commit messages.  And changelogs are for users, so I apply different rules there.

**Comment by hynek** [comment]
thanks @blueyed I’ve fixed some of the things!

**Pull request merged**